### ISSUE TITLE
Question: How to check for valid timezone.

### ISSUE THREAD

**Issue by brolaugh** [marnusw/date-fns-tz#143]
In a project that I'm working on were we're using version 1.1.1 of this repo we've got a a unit test that effectively does the following: 

```ts
expect(() => zonedTimeToUtc('2020-01-01T12:00:00.000Z', 'bad/timezone').toThrowError()
```

Now I upgraded to version  1.1.6 (latest at time of writing) and this test no longer passes. So I'm wondering if there's some way that I can check the validity of a timezone? 

Looking at the changes that has happened [between the two versions](https://github.com/marnusw/date-fns-tz/compare/16d088079596a6a0a2e75ddea29289ac747f9814..9ba274b58f8d06722f45035ad2e24f7b9f57bb93) as far as I understand if the timezone is bad it defaults to the clients current timezone.

**Comment by voliva** [comment]
I'm in the same situation - The fact that `date-fns-tz` is just swallowing the error and returning a valid date can be problematic.

**Comment by brolaugh** [comment]
@voliva not sure if this should be considered a hack or not but I did this and got the same error as before.

```
public buildUtcFromZonedDate(date: Date, timeZone: string): Date {
    Intl.DateTimeFormat(undefined, {timeZone});
    return zonedTimeToUtc(date, timeZone);
  }
``

**Comment by marnusw** [comment]
This was fixed in e51e3b66e. Functions that return dates will return `Invalid Date` for bad date / time zone inputs, and `format` functions throw a `RangeError`.

It won't make your particular test work as it did before, but you could test for an invalid date instead of an exception.
